tools/scan_includes.c

```diff
@@ -38,11 +38,8 @@ void *xmalloc(const size_t size)
 
 void options_add_file(char *fname)
 {
-	char *filename = strdup(fname);
-	if (!filename) {
-		perror("strdup");
-		exit(1);
-	}
+	char *filename = xmalloc(strlen(fname) + 1);
```

Comment:
I removed `strdup` because it generates warnings when compiling with `-std=c99` since `strdup` is not standard.
Note that the `malloc`d string is never `free`d, but the `strdup`d string wasn't `free`d either.

Reply:
As long as it doesn't build up recursively (it's just done to parse options at the start and used for the entire duration of the program) I think that's ok.

Reply:
I agree. Just commenting for clarity.